foundry/foundry.py

```diff
@@ -320,20 +332,31 @@ def get_citation(self) -> str:
         bibtex = f"@misc{{https://doi.org/{self.dc['identifier']['identifier']}{os.linesep}{bibtex}}}"
         return bibtex
 
-    def publish(self, foundry_metadata, data_source, title, authors, update=False,
-                publication_year=None, **kwargs,):
+    def publish_dataset(
+            self, foundry_metadata, title, authors, https_data_path=None, globus_data_source=None, update=False,
+            publication_year=None, test=False, **kwargs,):
         """Submit a dataset for publication
         Args:
             foundry_metadata (dict): Dict of metadata describing data package
-            data_source (string): Url for Globus endpoint
             title (string): Title of data package
             authors (list): List of data package author names e.g., Jack Black
                 or Nunez, Victoria
+            https_data_path (str): Path to the local dataset to publish to Foundry via HTTPS. Creates an HTTPS PUT
+                request to upload the data specified to a Globus endpoint (default is NCSA endpoint) before it is
```

Comment:
Do we need to cite the default is NCSA here? I'm looking for ways to shorten the docstring.

Reply:
Hm I guess I don't see how removing that parenthetical would significantly reduce the size of the docstring overall? But I do think it adds quick clarity to what this function is doing, such that a developer could just read the docstring and not the code to know where the data are being written to

Reply:
Looking again, I've got a different impression and am feeling kinda surprised I was advocating for less docs. 

Maybe I'd put the explanation of the two different modes for supplying the data earlier in the docstring so that people can understand how to get the desired behavior (e.g., upload via HTTPS) without having to read each argument.

Reply:
that sounds like a great addition! can do